OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -242,8 +242,25 @@ public override bool Tick(Actor self)
 				var cellRange = nearEnough.Length / 1024;
 				if (!containsTemporaryBlocker && (mobile.ToCell - destination.Value).LengthSquared <= cellRange * cellRange && mobile.CanStayInCell(mobile.ToCell))
 				{
-					path.Clear();
-					return null;
+					// Apply some simple checks to avoid giving up in cases where we can be confident that
+					// nudging/waiting/repathing should produce better results.
+
+					// Avoid fighting over the destination cell
+					var adjacentToDestination = path.Count < 2;
```

Comment:
Imho it would make sense to return early here and skip all the calculation below if the condition is met:
```
// Avoid fighting over the destination cell
if (path.Count < 2)
{
	path.Clear();
	return null;
}
```

Reply:
IMO the way I had it was a bit cleaner, but theres not much between the two so ok.